### ISSUE TITLE
Kernel upgrade to support new features

### ISSUE THREAD

**Issue by trevorturk** [heroku/roadmap#181]
### Required Terms

- [X] I agree to follow this project's [Code of Conduct](https://github.com/heroku/roadmap/blob/main/CODE_OF_CONDUCT.md)
- [X] I have read and accept the [Salesforce Program Agreement](https://www.salesforce.com/company/program-agreement/)

### What service(s) is this request for?

Heroku dynos

### Tell us about what you're trying to solve. What challenges are you facing?

I'd like to request that Heroku consider upgrading the build and deployment kernels to enable newer features such as `io_uring` and `epoll_pwait2`. 

This is a bit of a low-level / in-the-weeds sort of request, as discussed in https://github.com/socketry/io-event/issues/57 with @schneems; not a high-priority, but definitely a nice-to-have which should bring performance improvements to many users. 

In my case specifically, I'm using the latest Ruby with the Async suite of gems, which should have better performance if/when able to leverage `io_uring` and `epoll_pwait2`. 

Thanks for your consideration!

**Comment by friism** [comment]
Thanks for opening this, great improvement to track. We are actually in the process of updating the base operating system for both Common Runtime and Private Spaces which will also come with a more modern kernel. There'll be no separate upgrade required for builds since they run on the platform.

**Comment by jakeg** [comment]
When can we expect to be able to use kernels that support `io_uring` ?

**Comment by friism** [comment]
Coming very soon!

**Comment by jakeg** [comment]
That's good to here. But do you have an ETA, or a way we can get notified when it is released? I'm waiting on this before I can make a change in my app.

**Comment by friism** [comment]
The upgrade is now in progress. "Eco" and "Basic" common runtime dynos in the EU region and most of the US-East region should be seeing the new kernel version. We're rolling out the change fully in the coming weeks.

**Comment by elimchaysengSF** [comment]
FYI [we've started the complete rollout for both common runtime and private spaces dynos](https://devcenter.heroku.com/changelog-items/2713) - we will post an update on the changelog and here when the upgrade is fully rolled out! We're working our way through all of our dynos so it will likely be a few more weeks before all dynos have changed over. You can also check what Kernel version your app is on for yourself at anytime by running `% heroku run "uname -a" -a my-app-name`

**Comment by trevorturk** [comment]
Awesome news, thank you!

**Comment by jpcamara** [comment]
> FYI [we've started the complete rollout for both common runtime and private spaces dynos](https://devcenter.heroku.com/changelog-items/2713) - we will post an update on the changelog and here when the upgrade is fully rolled out! We're working our way through all of our dynos so it will likely be a few more weeks before all dynos have changed over. You can also check what Kernel version your app is on for yourself at anytime by running `% heroku run "uname -a" -a my-app-name`

@elimchaysengSF is there a particular output we should look for to know the updates have been applied? I was under the impression that kernel version alone did not mean the `epoll_pwait2` and `io_uring` were available.

**Comment by elimchaysengSF** [comment]
@jpcamara -- all apps are now upgraded and you run the command I posted above to see the new version. Regarding your question about the specifics of those features, do you mind opening a support ticket so one of our engineers can help answer? Thanks!

**Comment by elimchaysengSF** [comment]
This upgrade is now complete. All applications have been upgraded and the change has been stabilized in production. You can reference this [changelog](https://devcenter.heroku.com/changelog-items/2866) as well.
